### PR TITLE
Optimize responsive layout in Perils grid with ResizeObserver instead of window-based media queries

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

- Refactor how `Perils` component implements responsive number of column
- Commented why current solution uses JS responsive logic instead of usual CSS solutions

New solution uses container size instead of window size. This works better in terms debugger where we have 2 perils blocks size by size and window-based size is way off

**Performance effect**
~5% reduction in product page scripting time

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4601?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4601** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4601?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4593** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4593?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4592** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4592?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4598** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4598?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alebedev and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/components/Perils/Perils.tsx [hunk 1]
```diff
@@ -29,38 +29,39 @@ type Props = {
...
-            <PerilsAccordion key={`${index}-${peril.title}`} peril={peril} />
+    <PerilsAccordionGrid ref={containerRef}>
+      {numOfColumns == 0
+        ? null // We don't want to render anything before sizing to prevent rendering rework and content jump
+        : getPerilColumns(perilItems, numOfColumns).map((perils, index) => (
+            <PerilColumnFlex key={index}>
+              {perils.map((peril) => (
+                <PerilsAccordion key={`${index}-${peril.title}`} peril={peril} />
+              ))}
+            </PerilColumnFlex>
```
Comment (guilhermespopolin) [inline apps/store/src/components/Perils/Perils.tsx:50]: Can we maybe extract this into a different component (inside the same file) in a attempt to make this easier to read?

**Review by guilhermespopolin — APPROVED** [review]
Super nice refactoring. Didn't know about ResizeObserver

File: apps/store/src/components/Perils/Perils.tsx (hunk 1 above)
Comment (alebedev) [reply apps/store/src/components/Perils/Perils.tsx:50]: I think it's probably solved by refactoring further in this PR stack

**Comment by alebedev** [comment]
### Merge activity

* **Aug 19, 3:12 AM EDT**: @alebedev started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4601).

**Pull request merged**